src/scraper/acg-rip.ts

```diff
@@ -0,0 +1,156 @@
+    }
+
+    public async executeMainTask(pageNo?: number): Promise<{ items: Array<Item<number>>; hasNext: boolean; }> {
+        try {
+            let listPageUrl = AcgRipScraper._host;
+            if (pageNo) {
+                listPageUrl += `/page/${pageNo}`;
+            }
+            logger.info('execute_main_task', { pageNo });
+            const resp = await Axios.get(listPageUrl);
+
+            const $ = cheerio.load(resp.data);
+            const trList = Array.from($('table.post-index > tbody > tr'));
+            let items: Array<Item<number>> = [];
+            trList.forEach(tr => {
+                const item = new Item<number>();
+
+                // type
+                item.type = new ItemType<number>();
+                item.type.id = 1;  // hard coded type 1 as anime
+                item.type.name = '动画';
+
+                item.uri = $('td.title a', tr).last().attr('href').trim();
+                item.id = this.getIdFromUri(item.uri);
+
+                items.push(item);
+            });
+            let newIds = await this._store.filterItemNotStored(items.map(item => item.id));
+            let newItems = items.filter(item => {
+                return newIds.includes(item.id);
+            });
+            return { hasNext: newIds.length === items.length && newIds.length > 1, items: newItems };
+        } catch (e) {
+            if (e.code !== 'ETIMEDOUT') {
+                captureException(e);
+            }
+            logger.warn('execute_main_task_exception', {
+                code: e.code,
+                error_message: e.message,
+                line: '80',
+                page_no: pageNo,
+                stack: e.stack
+            });
+            return null;
+        }
+    }
+    public async executeSubTask(item: Item<number>): Promise<number> {
+        let statusCode = -1;
+        try {
+            const subTaskUrl = `${AcgRipScraper._host}${item.uri}`;
+            logger.info('execute_sub_task', { item });
+            const resp = await Axios.get(subTaskUrl);
+            statusCode = resp.status;
+
+            const $ = cheerio.load(resp.data);
+            const panels = $('.container .panel.post-show-content');
+            const sidePanel = $('.container .side-panel');
+            item.title = panels.eq(0).find('.panel-heading').text().trim();
+            item.publisher = new Publisher<number>();
+            const pub = sidePanel.eq(0).find('.panel-body dd > a[href]');
+            const pubId = pub.attr('href');
+            item.publisher.name = pub.text().trim();
+            item.publisher.id = this.getPublisherIdFromUri(pubId);
+            item.timestamp = new Date(Number(sidePanel.eq(0).find('time').attr('datetime')) * 1000);
+            item.torrent_url = AcgRipScraper._host + panels.eq(0).find('.panel-body a.btn').attr('href');
+            item.magnet_uri = await getMagnetUri(item.torrent_url);
+            item.files = await getMediaFiles(item.torrent_url);
```

Comment:
Can we avoid downloading the torrent file twice? you can just use one parse torrent to get content and magnet